Source/src/core/GameObject.cpp

```diff
@@ -528,3 +528,25 @@ Hachiko::GameObject* Hachiko::GameObject::GetFirstChildWithName(
 
     return nullptr;
 }
+
+Hachiko::GameObject* Hachiko::GameObject::FindDescendantWithName(const std::string& child_name) const
+{
+    for (GameObject* child : children)
+    {
+        if (child->name == child_name)
+        {
+            return child;
+        }
+    }
+
+    for (GameObject* child : children)
+    {
+        GameObject* found_child = child->FindDescendantWithName(child_name);
+        if (found_child != nullptr)
+        {
+            return found_child;
+        }
+    }
+
+    return nullptr;
+}
```

Comment:
Is there any reason for iterating two times over the children? if not, the content could be in just one for loop, I think is better for performance and for reading the code